runtime/src/cryptographic_mailbox.rs

```diff
@@ -0,0 +1,178 @@
+                }
+            }
+        }
+    }
+}
+
+#[derive(Default)]
+struct KeyUsageInfo {
+    key_id: u32,
+    #[allow(unused)]
+    counter: u64,
+}
+
+impl KeyUsage {}
+
+const UNENCRYPTED_CMK_SIZE_BYTES: usize = 80;
+
+#[repr(C)]
+#[derive(Clone, FromBytes, Immutable, IntoBytes, KnownLayout)]
+struct UnencryptedCmk {
+    version: u16,
+    length: u16,
+    key_usage: u8,
+    id: [u8; 3],
+    usage_counter: u64,
+    key_material: [u8; CMK_MAX_KEY_SIZE_BITS / 8],
+}
+
+impl UnencryptedCmk {
+    #[allow(unused)]
+    fn key_id(&self) -> u32 {
+        self.id[0] as u32 | ((self.id[1] as u32) << 8) | ((self.id[2] as u32) << 16)
+    }
+}
+
+#[repr(C)]
+#[derive(Clone, FromBytes, Immutable, IntoBytes, KnownLayout)]
+struct EncryptedCmk {
+    domain: u32,
+    domain_metadata: [u8; 16],
+    iv: [u8; 12],
+    ciphertext: [u8; UNENCRYPTED_CMK_SIZE_BYTES],
+    gcm_tag: [u8; 16],
+}
+
+pub(crate) struct Commands {}
+
+impl Commands {
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    #[inline(never)]
+    pub(crate) fn status(drivers: &mut Drivers) -> CaliptraResult<MailboxResp> {
+        let len = drivers.cryptographic_usage_data.counters.len();
+        Ok(MailboxResp::CmStatus(CmStatusResp {
+            hdr: MailboxRespHeader::default(),
+            used_usage_storage: len as u32,
+            total_usage_storage: KEY_USAGE_MAX as u32,
+        }))
+    }
+
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    #[inline(never)]
+    pub(crate) fn import(drivers: &mut Drivers, cmd_bytes: &[u8]) -> CaliptraResult<MailboxResp> {
+        if cmd_bytes.len() > core::mem::size_of::<CmImportReq>() {
+            Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS)?;
+        }
+        let mut cmd = CmImportReq::default();
+        cmd.as_mut_bytes()[..cmd_bytes.len()].copy_from_slice(cmd_bytes);
+
+        let key_usage = CmKeyUsage::from(cmd.key_usage);
+
+        let valid = matches!((key_usage, cmd.input_size), (CmKeyUsage::AES, 32));
+        if !valid {
+            Err(CaliptraError::RUNTIME_CMB_INVALID_KEY_USAGE_AND_SIZE)?;
+        }
+
+        let _raw_key = &cmd.input[..cmd.input_size as usize];
+        // [TODO][CAP2]: we need to generate our internal key to encrypt the CMK
+        let mut unencrypted_cmk = UnencryptedCmk {
+            version: 1,
+            length: cmd.input_size as u16,
+            key_usage: key_usage as u32 as u8,
+            id: [0u8; 3],
+            usage_counter: 0,
+            key_material: [0u8; CMK_MAX_KEY_SIZE_BITS / 8],
+        };
+
+        // keep track of counters for AES
+        if matches!(key_usage, CmKeyUsage::AES) {
+            unencrypted_cmk.id = drivers.cryptographic_usage_data.add_counter()?;
+        }
+
+        let encrypted_cmk = EncryptedCmk {
+            domain: 0,
+            domain_metadata: [0u8; 16],
+            iv: [0u8; 12],
+            ciphertext: transmute!(unencrypted_cmk),
```

Comment:
nit: I know there's no key material in here yes, but it makes me a bit nervous to return the plaintext cmk here. Seems a bit error prone when adding code later since we have to remember to fix it.

It might make sense to return an error here for now until we have the CMK encryption code set up.

Reply:
I'll just return 0s for now instead.